src/clipping/sutherlandhodgman.jl

```diff
@@ -14,6 +14,40 @@ The Sutherland-Hodgman algorithm for clipping polygons.
 """
 struct SutherlandHodgman <: ClippingMethod end
 
-function clip(poly::Polygon, other::Geometry, ::SutherlandHodgman)
-  @error "not implemented"
+function clip(poly::Polygon{Dim,T}, other::Polygon, ::SutherlandHodgman) where {Dim,T}
+  vother = vertices(other)
+  n = length(vother)
+  v = vertices(poly)
+
+  for i in 1:n
+    l₁ = Line(vother[i], vother[mod1(i+1, n)])
+
+    m = length(v)
+    newv::Vector{Point{Dim,T}} = []
```

Comment:
I think this is simpler and more "julian":
```suggestion
    newv = Point{Dim,T}[]
```